juju/client/proxy/factory.py

```diff
@@ -0,0 +1,39 @@
+from abc import ABC, abstractmethod
+from juju.client.proxy.kubernetes.proxy import KubernetesProxy
+
+class ProxyTypeNotDefinedError(Exception):
+    pass
+
+class UnknownProxyTypeError(Exception):
+    pass
+
+class Factory(ABC):
+
+    @abstractmethod
+    def proxy_from_config(self, conf):
+        raise NotImplementedError()
+
+class JujuDataFactory(Factory):
+
+    def proxy_from_config(self, conf):
+        if conf is None:
+            return None
+
+        if "type" not in conf:
+            return None
+
+        proxy_type = conf['type']
+        if proxy_type != "kubernetes-port-forward":
+            raise UnknownProxyTypeError()
```

Comment:
You typically want to pass a message here, including the value that was incorrect. For example:

    raise ValueError('invalid proxy type {!r}'.format(proxy_type))

The `!r` format syntax prints the repr of the value, i.e., quoted string.